Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -740,3 +756,26 @@ public enum RestError: Error, CustomStringConvertible {
         }
     }
 }
+
+// URL Session extension
+extension RestRequest: URLSessionDelegate {
+    
+    /// URL session function to allow trusting certain URLs
+    public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+        let method = challenge.protectionSpace.authenticationMethod
+        let host = challenge.protectionSpace.host
+        switch (method, host) {
+        case (NSURLAuthenticationMethodServerTrust, self.url):
+            #if MAC_OS_X_VERSION_10_6
```

Comment:
@ShihabMehboob A few points regarding the issue that `challenge.protectionSpace.serverTrust` is not available on Linux:

- When was `challenge.protectionSpace.serverTrust` introduced to the macOS API? Is it only available on macOS VERSION_10_6 and later? If it was recently introduced, then I am wondering what APIs were used prior to this addition in order to achieve the same goal/behavior we need.

- Is there another way (e.g. different API methods) for us to achieve the same goal and be platform neutral? It may be worth to poke around the API docs and see if there are any alternatives.

- If in the end, this is the only viable solution, I recommend we log warnings in this section of the code whenever we are running on a platform that does not have yet the necessary API implementations... but I am hoping there is a platform neutral solution.

- Either way, we should notify the foundation folks about the discrepancy in the API (macOS vs linux). I recommend you share these details... @ianpartridge or @pushkarnk can probably be the first point of contact?